src/osd/OSD.h

```diff
@@ -1691,7 +1691,8 @@ class OSD : public Dispatcher,
 
   void handle_osd_map(class MOSDMap *m);
   void _committed_osd_maps(epoch_t first, epoch_t last, class MOSDMap *m);
-  void trim_maps(epoch_t oldest);
+  // return true if trimming actually happend, false otherwise
+  bool trim_maps(epoch_t oldest);
```

Comment:
Should we update existing users?
```
-    trim_maps(m->cluster_osdmap_trim_lower_bound);
-    pg_num_history.prune(superblock.get_oldest_map());
+    if (bool trimmed = trim_maps(m->cluster_osdmap_trim_lower_bound);
+        trimmed) {
+     pg_num_history.prune(superblock.get_oldest_map());
+    }
```

Either update it to avoid possible compilation return type warnings